test/e2e/common_suite_test.go

```diff
@@ -73,38 +101,69 @@ func (tc *testCase) run() {
 					t.Fatal(err)
 				}
 			}
-
-			if err = client.Resources().Create(ctx, tc.pod); err != nil {
-				t.Fatal(err)
+			if tc.job != nil {
+				if err = client.Resources().Create(ctx, tc.job); err != nil {
+					t.Fatal(err)
+				}
+				if err = wait.For(conditions.New(client.Resources()).JobCompleted(tc.job), wait.WithTimeout(WAIT_JOB_RUNNING_TIMEOUT)); err != nil {
+					t.Log(err)
+				}
 			}
-
-			if err = wait.For(conditions.New(client.Resources()).PodRunning(tc.pod), wait.WithTimeout(WAIT_POD_RUNNING_TIMEOUT)); err != nil {
-				t.Fatal(err)
+			if tc.pod != nil {
```

Comment:
FYI, I am not reviewing the `if tc.pod != nil` blocks because I assumed you kept the same old logic. Let me know otherwise.

Reply:
As all test cases don't need pod (t.pod) some might need tc.job or tc.deployment so for refactoring the test cases related to pods only we made a condition there.